chrome/browser/net/chrome_network_delegate.cc

```diff
@@ -86,6 +87,38 @@ namespace {
 
 bool g_access_to_all_files_enabled = false;
 
+const std::regex RE_JS("\\.js(?:\\?.+)*$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_CSS("\\.css(?:\\?.+)*$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_IMAGE("\\.(?:gif|png|jpe?g|bmp|ico)(?:\\?.+)*$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_FONT("\\.(?:ttf|woff)(?:\\?.+)*$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_HTML("\\.html?(?:\\?.+)*$", std::regex_constants::ECMAScript | std::regex_constants::icase);
```

Comment:
I'm not sure I understand the regular expressions correctly. 

`(?:\?.+)` will match `?` followed by one or more char. But `(?:\?.+)*$` will match 0 or many of ('?' followed by one or more char) until the end of string. I don't think you need `*$` in any of the regexp, but instead you want  `(?:\?.+)?` to indicate the optional query.

Reply:
yes, we can remove `*` before `$`.
but i believe `$` at the end will not hurt.

Reply:
I'd be interested to know if having the `$` at the end has an impact on performance. 

I'm personally of the opinion that if you don't need it, you should remove it.

Reply:
i believe it makes difference only in multiline mode and all our lines here are just one-lines.

I'm not JS expert but quick googling leads to https://stackoverflow.com/a/5921779/1103561 that suggests to use them. However i'd say it can theoretically depend on V8 (regexp engine) version so i'd vote for writing quick test in c++ (libadblockplus) measuring time for doing 10^4 regexp matching for 2 patterns: with `$` and without it.

I bet it's more or less the same for single-lines.

Reply:
Fair enough.

Reply:
Actually it's executed in C++, not in JS, but it's not important. I think it makes sense to drop `$` because we don't need to scan the string till the end. What we really need is to find `?` because it's usually used to separate *query* from *path* and we assume that at the end of the path there is a file extension, what we are interested in and we are not interested in the query component.

Reply:
Oh, yes. I was testing regexp in JS so i was confused. Obviously it works in C++.